examples/tfengine/generated/team/project_secrets/main.tf

```diff
@@ -52,6 +52,9 @@ module "project" {
   # It is a no-op when Kubernetes Engine API is not enabled in the project.
   grant_services_security_admin_role = true
   activate_apis                      = ["secretmanager.googleapis.com"]
+  labels = {
```

Comment:
Can we also add a label in the project block (https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/blob/master/examples/tfengine/modules/team.hcl#L36) and make sure that label value can also get propagated here?

Reply:
I tested explicitly adding labels in the block. They were correctly merged with the labels defined at root.
There was no instance where explicitly defined label was not accompanied by one from the root.

Do we want to reference labels defined in a parent scope, and not only current or root scope?

Reply:
There is an issue in the current implementation.

Adding labels map in the project block propagates them down to the appropriate project template.
However, that merged map also propagates to subsequent project templates in the recipe that should not have inherited the map.

Reply:
Analyzing `examples/tfengine/team.hcl` with labels added to the `project_secrets`
https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/blob/d955b6b5d8a58a8589d6716e325c1eab2e3cf211/examples/tfengine/modules/team.hcl#L36-L41

In `recipes/project.hcl`, `project` template configures `flatten` for the `project` block.
https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/blob/d955b6b5d8a58a8589d6716e325c1eab2e3cf211/templates/tfengine/recipes/project.hcl#L165-L170

Next, the elevated fields (including `labels`) are merged onto `templateInfo.Data`, which is merged back onto local `data` map that was earlier merged with `conf.Data`.
https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/blob/d955b6b5d8a58a8589d6716e325c1eab2e3cf211/internal/tfengine/tfengine.go#L166-L180

After that, the `labels` map of the root object contains unwanted project labels.

Reply:
58fb136 adds a label in the project block and shows the proper propagation of that label.
57a1bbc makes a copy of root data map, before merging local data maps.